tests/test_download.py

```diff
@@ -168,81 +172,188 @@ def test_download_url_to_tempfileobj_and_urls(self):
+  #   Use some known URLs on the net to ensure that TUF download checks SSL
+  #   certificates appropriately.
+  #   """
+  #   # We should never get as far as the target file download itself, so the
+  #   # length we pass to safe_download and unsafe_download shouldn't matter.
+  #   irrelevant_length = 10
+  #
+  #   for bad_url in [
+  #       'https://expired.badssl.com/', # expired certificate
+  #       'https://wrong.host.badssl.com/', ]: # hostname verification fail
+  #
+  #     with self.assertRaises(requests.exceptions.SSLError):
+  #       download.safe_download(bad_url, irrelevant_length)
+  #
+  #     with self.assertRaises(requests.exceptions.SSLError):
+  #       download.unsafe_download(bad_url, irrelevant_length)
+
+
 
 
 
   def test_https_connection(self):
+    """
+    Try various HTTPS downloads using trusted and untrusted certificates with
+    and without the correct hostname listed in the SSL certificate.
+    """
     # Make a temporary file to be served to the client.
     current_directory = os.getcwd()
     target_filepath = self.make_temp_data_file(directory=current_directory)
-    target_data = None
-    target_data_length = 0
 
     with open(target_filepath, 'r') as target_file_object:
-      target_data = target_file_object.read()
-      target_data_length = len(target_data)
-
-    # Launch an https server (serves files in the current dir).
-    port = random.randint(30000, 45000)
-    command = ['python', 'simple_https_server.py', str(port)]
-    https_server_process = subprocess.Popen(command, stderr=subprocess.PIPE)
-
-    # NOTE: Following error is raised if delay is not applied:
-    #    <urlopen error [Errno 111] Connection refused>
-    time.sleep(1)
-
-    junk, relative_target_filepath = os.path.split(target_filepath)
-    https_url = 'https://localhost:' + str(port) + '/' + relative_target_filepath
+      target_data_length = len(target_file_object.read())
+
+    # These cert files provide various test cases:
+    # good:    A valid cert from an older generation of test_download.py tests.
+    # good2:   A valid cert made simultaneous to the bad certs below, with the
+    #          same settings otherwise, tested here in case the difference
+    #          between the way the new bad certs and the old good cert were
+    #          generated turns out to matter at some point.
+    # bad:     An otherwise-valid cert with the wrong hostname. The good certs
+    #          list "localhost", but this lists "notmyhostname".
+    # expired: An otherwise-valid cert but which is expired (no valid dates
+    #          exist, fwiw: startdate > enddate).
+    good_cert_fname = os.path.join('ssl_certs', 'ssl_cert.crt')
+    good2_cert_fname = os.path.join('ssl_certs', 'ssl_cert_2.crt')
+    bad_cert_fname = os.path.join('ssl_certs', 'ssl_cert_wronghost.crt')
+    expired_cert_fname = os.path.join('ssl_certs', 'ssl_cert_expired.crt')
+
+    # Launch three https servers (serve files in the current dir).
```

Comment:
`s/three/four/`, and please explain why there are two good ones.

Reply:
Thanks for the catch. There's an explanation 13 lines up, [here]( https://github.com/theupdateframework/tuf/pull/781/files/b163caa29b37f2794201cbec43be2f6697ade745#diff-e3c1e2065ceb55ca960d7bc2bd8de44cR240). I'll adjust this, though, for clarity.